commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MeanTest.java

```diff
@@ -57,31 +60,108 @@ void testMean(double[] values, double expected) {
-                8.2, 10.3, 11.3, 14.1, 9.9, 12.2, 12.0, 12.1, 11.0, 19.8, 11.0, 10.0, 8.8,
-                9.0, 12.3}, 12.404545454545454),
-            Arguments.of(new double[] {-0.0, +0.0}, 0.0),
-            Arguments.of(new double[] {0.0, -0.0}, 0.0),
-            Arguments.of(new double[] {0.0, +0.0}, 0.0),
-            Arguments.of(new double[] {0.001, 0.0002, 0.00003, 10000.11, 0.000004},
-                2000.0222468),
-            Arguments.of(new double[] {10E-50, 5E-100, 25E-200, 35.345E-50},
-                1.133625E-49),
+            Arguments.of(new double[] {Double.MAX_VALUE, -Double.MAX_VALUE}),
+            Arguments.of(new double[] {10, 8, 13, 9, 11, 14, 6, 4, 12, 7, 5}),
+            Arguments.of(new double[] {8.04, 6.95, 7.58, 8.81, 8.33, 9.96, 7.24, 4.26, 10.84, 4.82, 5.68}),
+            Arguments.of(new double[] {9.14, 8.14, 8.74, 8.77, 9.26, 8.10, 6.13, 3.10, 9.13, 7.26, 4.74, 7.46, 6.77, 12.74, 7.11, 7.81, 8.84, 6.08, 5.39, 8.15, 6.42, 5.73}),
+            Arguments.of(new double[] {8, 8, 8, 8, 8, 8, 8, 19, 8, 8, 8}),
+            Arguments.of(new double[] {6.58, 5.76, 7.71, 8.84, 8.47, 7.04, 5.25, 12.50, 5.56, 7.91, 6.89}),
+            Arguments.of(new double[] {0, 0, 0.0}),
+            Arguments.of(new double[] {1, -7, 6}),
+            Arguments.of(new double[] {1, 7, -15, 3}),
+            Arguments.of(new double[] {2, 2, 2, 2}),
+            Arguments.of(new double[] {2.3}),
+            Arguments.of(new double[] {3.14, 2.718, 1.414}),
+            Arguments.of(new double[] {12.5, 12.0, 11.8, 14.2, 14.9, 14.5, 21.0, 8.2, 10.3, 11.3, 14.1, 9.9, 12.2, 12.0, 12.1, 11.0, 19.8, 11.0, 10.0, 8.8, 9.0, 12.3}),
+            Arguments.of(new double[] {-0.0, +0.0}),
+            Arguments.of(new double[] {0.0, -0.0}),
+            Arguments.of(new double[] {0.0, +0.0}),
+            Arguments.of(new double[] {0.001, 0.0002, 0.00003, 10000.11, 0.000004}),
+            Arguments.of(new double[] {10E-50, 5E-100, 25E-200, 35.345E-50}),
+            Arguments.of(new double[] {Double.MAX_VALUE, Double.MAX_VALUE}),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -Double.MAX_VALUE}),
+            Arguments.of(new double[] {Double.MAX_VALUE, 1}),
+            Arguments.of(new double[] {-Double.MAX_VALUE, 1, 1}),
+            Arguments.of(new double[] {-Double.MAX_VALUE, -1, 1}),
+            Arguments.of(new double[] {Double.MAX_VALUE, -1}),
+            Arguments.of(new double[] {Double.MAX_VALUE, -Double.MAX_VALUE}),
+            Arguments.of(new double[] {1, -Double.MAX_VALUE}),
+            Arguments.of(new double[] {1, 1, 1, -Double.MAX_VALUE})
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "testMeanNonFinite")
+    void testMeanNonFinite(double[] values, double expected) {
+        Mean mean = Mean.create();
+        for (double value : values) {
+            mean.accept(value);
+        }
+        TestHelper.assertEquals(expected, mean.getAsDouble(), 2, () -> "mean non-finite");
```

Comment:
Since these are non-finite then we can use `Assertions.assertEquals` without the ulp tolerance. These should be exact. Note this is allowed:
```java
TestHelper.assertEquals(Double.POSITIVE_INFINITY, Double.MAX_VALUE, 1, () -> "oops");
```
We should update the TestHelper so it can detect non-finite inputs and uses binary equality. We really only want to use a ULP tolerance between finite numbers. This is a limitation of the Precision class which uses a count of the representable doubles between two values. This does not lend itself well to testing values that are close to infinity, e.g. MAX_VALUE, like we do in this class.

The current test suite for other statistics works if using this:
```java
static void assertEquals(double expected, double actual, int ulps, Supplier<String> msg) {
    // Require strict equivalence of non-finite values
    if (Double.isFinite(expected) && Double.isFinite(actual)) {
        Assertions.assertTrue(Precision.equals(expected, actual, ulps),
            () -> expected + " != " + actual + " within " + ulps + " ulp(s): " + msg.get());
    } else {
        Assertions.assertEquals(expected, actual, msg);
    }
}
```
But some of the MeanTest cases fail because the `computeExpected` uses a DECIMAL64 math context for the divide which rounds the value above Double.MAX_VALUE. Then the conversion to double creates infinity. If using DECIMAL128 then the MeanTest works.